src/util/epochguard.h

```diff
@@ -40,6 +40,9 @@ class LOCKABLE Epoch
     Epoch() = default;
     Epoch(const Epoch&) = delete;
     Epoch& operator=(const Epoch&) = delete;
+    Epoch(Epoch&&) = delete;
+    Epoch& operator=(Epoch&&) = delete;
```

Comment:
The `&&` constructors should already be deleted, no?

> the presence of a user-defined destructor, copy-constructor, or copy-assignment operator prevents implicit definition of the move constructor and the move assignment operator

From the page you linked (https://en.cppreference.com/w/cpp/language/rule_of_three#Rule_of_five)

And quoting from your error message this should be the case: "... has a user-declared copy assignment operator".

Reply:
Ah ok, the other pages you linked to seem to always prefer all 5 (or zero)